### PR TITLE
Gpu 1

### PR DESCRIPTION
**Pull request by ausaafnabi**
- Please Check the Pearson Function

### CODE REVIEW

File: mtrx.R [hunk 1]
```diff
@@ -1,121 +1,197 @@
...
-  return(data)
-}
+process_batch <- function(count_matrix, first_index, second_index, batch_size, metric) {
+    if (first_index == second_index) {
+        if (metric == 'kendall') {
+            fn_name <- "matrix_Kendall_distance_same_block"
+        } else if(metric =='euclidean') {
+            fn_name <- "matrix_Euclidean_distance_same_block"
+        } else if(metric =='pearson') {
+            fn_name <- "matrix_Pearson_distance_different_blocks" #for not different block
```
Comment (akhtyamovpavel) [inline mtrx.R]: Why here is `different_blocks`?

File: main.cu [hunk 2]
```diff
@@ -84,6 +84,87 @@ __global__ void Rkendall_gpu_atomic_float_different_blocks(float* array, float*
...
+
+  for (int col1_num = 0; col1_num < m; ++col1_num) {
+    for (int col2_num = 0; col2_num < m_b; ++col2_num) {
+      float* col1 = array + n * col1_num;
+      float* col2 = array2 + n * col2_num;
+      if (row < m) {
+        if(col2[row]==0.0){
+          
+          float diff = col1[row] - col2[row];
+          diff = diff * diff;
```
Comment (akhtyamovpavel) [inline main.cu:178]: Have you checked this formula?

**Review by akhtyamovpavel — CHANGES_REQUESTED** [review]
Hello!

I have remained two comments. Could you watch it, please?

File: main.cu [hunk 3]
```diff
@@ -130,40 +130,36 @@ __global__ void RpearsonCorr_gpu_atomic_float_same_block(
...
-	int row = blockIdx.y*blockDim.y +threadIdx.y;
-	int row2 = blockIdx.x*blockDim.x +threadIdx.x;
-	float epsilon=0.01;
-	for (int col1_num=0;col1_num<m;++col1_num){
-	float num = 0;
-	float sum1 = 0;
-	float sum2 = 0;
-	float dist = 0;
+
+__global__ void RpearsonCorr3_gpu_atomic_float_different_blocks(float* array, float* array2, const int n, const int m, const int m_b, float* r1,float *r2, float* r3 ){
```
Comment (akhtyamovpavel) [inline main.cu]: Please, name variables not as `r1`, `r2` and `r3`

File: main.cu [hunk 4]
```diff
@@ -130,40 +130,36 @@ __global__ void RpearsonCorr_gpu_atomic_float_same_block(
...
+    int row = blockIdx.x*blockDim.x +threadIdx.x;
+    for (int col1_num=0;col1_num<m;++col1_num){
 	for(int col2_num=0;col2_num<m_b;++col2_num){
```
Comment (akhtyamovpavel) [inline main.cu]: You have doubled the code calculation. Please, watch it carefully

**Review by akhtyamovpavel — CHANGES_REQUESTED** [review]

File: main.cu (hunk 3 above)
Comment (ausaafnabi) [reply main.cu]: ok

File: main.cu (hunk 4 above)
Comment (ausaafnabi) [reply main.cu]: I'll fix this.

**Pull request merged**